### ISSUE TITLE
V9 Backups every few seconds on initial setup

### ISSUE THREAD

**Issue by bobbitchin** [Yenthe666/auto_backup#41]
While first configuring, I set the scheduler to run on weekdays, and to repeat missed.  I noticed the next scheduled run date was in the past (2010), but didn't think much of it.  After saving the settings, I hit the run manually button once and it started generating backups about one every 10 or 15 seconds.  I tried to change the repeat missed option and save the record, but kept getting an error message that the cron job was executing.  I finally got it to stop by stopping the odoo server, then hitting the save button, then restarting the odoo server.

**Comment by sdaguet** [comment]
Have the same problem. Looking at the odoo log, see

2016-05-04 09:30:20,782 4534 INFO None werkzeug: 127.0.0.1 - - [04/May/2016 09:30:20] "POST /xmlrpc/db HTTP/1.1" 200 -
2016-05-04 09:30:20,785 4534 INFO None openerp.service.db: DUMP DB: XXXX format zip

Then sftp file transfer and one minute later: 
2016-05-04 09:31:11,831 4530 ERROR ? openerp.service.server: Worker (4539) timeout
2016-05-04 09:31:11,896 4530 ERROR ? openerp.service.server: Worker (4539) timeout
2016-05-04 09:31:11,987 9855 INFO ? openerp.service.server: Worker WorkerCron (9855) alive

And another backup ...
2016-05-04 09:31:52,053 4533 INFO None werkzeug: 127.0.0.1 - - [04/May/2016 09:31:52] "POST /xmlrpc/db HTTP/1.1" 200 -
2016-05-04 09:31:52,058 4533 INFO None openerp.service.db: DUMP DB: XXXX format zip

**Comment by Yenthe666** [comment]
@bobbitchin and @sdaguet I believe this is because of the combination scheduled run date that is so far in the past and the repeat missed being true.
Repeat missed means it will start doing all 'assumed' missed automated schedules, meaning it would execute your schedule for 6 years worth of back-ups.

Either change the scheduled run date or tick off repeat missed, that should fix it.

**Comment by sdaguet** [comment]
@Yenthe666 I confirm repeat missing is not checked, reinstalled the module yesterday and put a yesterday date as start date.
Can work during days without any problem, and sometime ... it start to produce backup every few minutes and full the disk ... for example last week

28_04_2016_03_14_36_xxx
28_04_2016_03_16_29_xxx
28_04_2016_03_19_29_xxx
28_04_2016_03_21_00_xxx
28_04_2016_03_22_40_xxx

**Comment by alaa9jo** [comment]
I noticed something like that but not sure if this is the same issue or not

After installing the module today, I configured next schedule to be **18 July 2016 00:00:00** and repeat missed was not checked then clicked on save, waited few seconds then clicked on Run manually , then I checked on the folder and found two backups:

```
> ls --full-time
2016-07-18 11:34:04.523857799 18_07_2016_11_34_00_test.zip
2016-07-18 11:34:11.519857799  18_07_2016_11_34_08_test.zip
```

My guess is: the second backup file is generated from clicking on Run manually, but the first one is generated from clicking on Save, maybe it has checked if next scheduled is in the past or not, if it's then there is a missed call and it runs automatically even if repeat missing is not checked

**Comment by TimLai125** [comment]
Hi @Yenthe666 

I have a similar issue on v10 that the backup scheduler keep executing.
The "repeat missed" option is not enable. May I have any suggestions regarding my situation.
Following is part of the Odoo log.
```
2017-08-22 02:22:03,680 405 DEBUG test odoo.addons.base.ir.ir_cron: Starting job `Backup Scheduler`.
2017-08-22 02:22:03,700 405 DEBUG test odoo.modules.registry: Multiprocess load registry signaling: [Registry: 67] [Cache: 2791]
2017-08-22 02:22:03,704 405 INFO test odoo.modules.loading: loading 1 modules...
2017-08-22 02:22:03,704 405 DEBUG test odoo.modules.loading: loading module base (1/1)
2017-08-22 02:22:03,726 405 INFO test odoo.modules.loading: 1 modules loaded in 0.02s, 0 queries
2017-08-22 02:22:03,814 405 DEBUG test odoo.modules.loading: Updating graph with 126 more modules
2017-08-22 02:22:03,815 405 INFO test odoo.modules.loading: loading 127 modules...
2017-08-22 02:22:03,815 405 DEBUG test odoo.modules.loading: loading module base_technical_features (2/127)
...
2017-08-22 02:22:04,027 405 DEBUG test odoo.modules.loading: loading module stock_shipment_schedule_report (127/127)
2017-08-22 02:22:04,028 405 INFO test odoo.modules.loading: 127 modules loaded in 0.21s, 0 queries
2017-08-22 02:22:04,082 405 WARNING test odoo.fields: Field account.invoice.line.standard_price: parameter 'digits_compute' is no longer supported; use 'digits' instead.
2017-08-22 02:22:04,264 405 INFO test odoo.modules.loading: Modules loaded.
2017-08-22 02:22:04,472 405 DEBUG test odoo.modules.registry: Multiprocess signaling check: [Registry - 67 -> 67] [Cache - 2791 -> 2791]
2017-08-22 02:22:04,474 405 DEBUG test odoo.addons.base.ir.ir_cron: cron.object.execute(u'test', 1, '*', u'db.backup', u'action_backup_all')
2017-08-22 02:22:04,476 405 INFO test odoo.service.db: DUMP DB: test format zip
2017-08-22 02:22:12,070 405 INFO test odoo.addons.auto_backup.models.db_backup: Starting database backup: sftp://***************************
2017-08-22 02:22:12,070 405 DEBUG test odoo.addons.auto_backup.models.db_backup: Trying to connect to sftp://%(username)s@%(host)s:%(port)d
2017-08-22 02:22:12,299 405 INFO ? paramiko.transport: Connected (version 2.0, client OpenSSH_7.2p2)
2017-08-22 02:22:12,947 405 INFO ? paramiko.transport: Authentication (publickey) successful!
2017-08-22 02:22:13,441 405 INFO test paramiko.transport.sftp: [chan 0] Opened sftp connection (server version 3)
```

**Comment by Yenthe666** [comment]
Hmm I honestly have no idea where it comes from. I haven't had any reports about it except from you two. I've got multiple instances running this module too and never faced it. So I'm not sure, sorry..

**Comment by TimLai125** [comment]
Hi @Yenthe666 

I just realize that the backup dump files that the server created are unable to be extracted and the files are made in 5 minutes interval.
However, I know that the odoo cron task should not have a timeout limit but it seems that the cron task might be killed due to the Odoo server timeout.
Please let me know if you have any opinion regarding the timeout settings and the cron task.

**Comment by silentinfotech** [comment]
@Yenthe666 

We've experienced similar issue on one of our production servers.
Trying out multiple ways to replicate it onto dev server, we've following things as conclusion:
- Odoo executes the scheduled actions on workers & its execution is controlled by config param 'limit_time_real' (default value 120 seconds).
- Next execution date of scheduled action gets updated after successful execution of the task function.
- If the backup task is not completed within the configured limit time - worker will be killed. Which results into partial backup file (not extractable). And next execution time is not updated.
- Hence when ir_cron looks up for eligible schedule action - backup task will be elegible for execution
- This repeats untill successful backup is done within configured worker timeout OR it repeats for ever.

Common Cause :
- DB size is too large to be backed up within default configured time out.
- Running odoo on VPS might result in situation where resources(disk IO, CPU) are not fairly available at the time of backup. This result in success after few times of execution.

Possible solution :
- Can increase the timeout limit to some reasonable value based on size of DB backup.

Bottom line : This can cause problem to any such scheduled action where the task is not completed withi n defined limit_time_real

**Comment by Yenthe666** [comment]
Hi @ghost,

Thanks a lot for this awesome analysis with a lot of information!
There is not much that I can do from the perspective of the module itself to fix this. As this can take ages to figure out and debug though (as you can notice by the create date of this issue) I've decided to add it in all the `readme.md` files. I believe this is the best I can do to help others to not run into this issue.
V9: https://github.com/Yenthe666/auto_backup/commit/098092aea859a96185295e154428b1bb952d42a0
V10: https://github.com/Yenthe666/auto_backup/commit/bf3d9d0fab29b153ffc08f7cc26a766b5f83f3dc
V11: https://github.com/Yenthe666/auto_backup/commit/2da967e7ac58c25a5505283a7a6ef6de2c2397be

I believe I can now (finally) close this report as it is documented and user related. If you have anymore suggestions or would like to see this adressed in another way then let me know.
Thanks again!
